src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2820,31 +2820,42 @@ private TypeWithState VisitCallReceiver(BoundCall node)
             return receiverType;
         }
 
-        private VisitResult GetReturnType(MethodSymbol method)
+        private TypeWithState GetReturnTypeWithState(MethodSymbol method)
         {
-            var type = method.ReturnTypeWithAnnotations;
-            var rvalueType = TypeWithState.Create(type, GetAnnotations(method));
-            return new VisitResult(rvalueType, type);
+            return TypeWithState.Create(method.ReturnTypeWithAnnotations, GetRValueAnnotations(method));
         }
 
-        private FlowAnalysisAnnotations GetAnnotations(ParameterSymbol parameter)
+        private FlowAnalysisAnnotations GetRValueAnnotations(Symbol symbol)
         {
             // Annotations are ignored when binding an attribute to avoid cycles. (Members used
             // in attributes are error scenarios, so missing warnings should not be important.)
-            return IsAnalyzingAttribute ?
-                FlowAnalysisAnnotations.None :
-                parameter.FlowAnalysisAnnotations;
+            if (IsAnalyzingAttribute)
+            {
+                return FlowAnalysisAnnotations.None;
+            }
+
+            var annotations = symbol switch
+            {
```

Comment:
>{ [](start = 12, length = 1)

It is probably just me, but it feels like switch expressions formatted like that aren't readable in my opinion. It would be better if the brace was aligned with the ```symbol``` instead of the ``` var```.

Reply:
This formatting complies with what the IDE enforces (auto-format).
I don't have a strong preference either way, but to change this we should file an IDE issue.

---
In reply to: [295443839](https://github.com/dotnet/roslyn/pull/36444#discussion_r295443839) [](ancestors = 295443839)